pymeasure/instruments/attocube/anc300.py

```diff
@@ -29,7 +29,45 @@
                                                      extract_float,
                                                      extract_int,
                                                      extract_value)
-from pymeasure.instruments.validators import strict_discrete_set, strict_range
+from pymeasure.instruments.validators import (joined_validators,
+                                              strict_discrete_set,
+                                              strict_range)
+
+
+def strict_length(value, values):
+    if len(value) != values:
+        raise ValueError(
+            'Value {} does not have an appropriate length of {}'.format(
+            value, values))
+    return value
+
+
+def truncated_int_array(value, values):
+    """ validator function to check if the supplied value is an iterable-object
+    with appropriate values.
+    """
+    ret = []
+    for i, v in enumerate(value):
```

Comment:
is this check going to happen often/frequently? If yes, it might be good to check if it would not run faster to have the "happy case" return early, a la
```
if lower <= v <= upper:
    if float(v).is_integer
       return v
    else:
       raise...
else:
    return max(min(upper, v), lower)

Reply:
This will happen very seldom. Its a basic configuration setting which most people will never touch. I will quickly check the run-time and make according changes